pkg/operator/validation.go

```diff
@@ -0,0 +1,178 @@
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidPrestoConfig ensure all Presto* fields are valid if provided.
+func IsValidPrestoConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, isValidHostPort(cfg.PrestoHost, "prestoHost"))
+
+	if !cfg.PrestoUseTLS {
+		if cfg.PrestoUseClientCertAuth {
+			errs = append(errs, fmt.Errorf("prestoUseClientCertAuth cannot be set to true if prestoUseTLS is false"))
+		}
+
+		if len(cfg.PrestoCAFile) > 0 {
+			errs = append(errs, fmt.Errorf("prestoCAFile cannot be set if prestoUseTLS is false"))
+		}
+	}
+
+	if len(cfg.PrestoCAFile) > 0 {
+		if _, err := os.Stat(cfg.PrestoCAFile); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if (len(cfg.PrestoClientCertFile) > 0 && len(cfg.PrestoClientKeyFile) == 0) ||
+		(len(cfg.PrestoClientKeyFile) > 0 && len(cfg.PrestoClientCertFile) == 0) {
+		errs = append(errs, fmt.Errorf("prestoClientCertFile and prestoClientKeyFile must both be specified or neither specified"))
+	}
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidHiveConfig ensure all Hive* fields are valid if provided.
+func IsValidHiveConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, isValidHostPort(cfg.HiveHost, "hiveHost"))
+
+	if !cfg.HiveUseTLS {
+		if cfg.HiveUseClientCertAuth {
+			errs = append(errs, fmt.Errorf("hiveUseClientCertAuth cannot be set to true if hiveUseTLS is false"))
+		}
+
+		if len(cfg.HiveCAFile) > 0 {
+			errs = append(errs, fmt.Errorf("hiveCAFile cannot be set if hiveUseTLS is false"))
+		}
+	}
+
+	if len(cfg.HiveCAFile) > 0 {
+		if _, err := os.Stat(cfg.HiveCAFile); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if (len(cfg.HiveClientCertFile) > 0 && len(cfg.HiveClientKeyFile) == 0) ||
+		(len(cfg.HiveClientKeyFile) > 0 && len(cfg.HiveClientCertFile) == 0) {
+		errs = append(errs, fmt.Errorf("hiveClientCertFile and hiveClientKeyFile must both be specified or neither specified"))
+	}
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidKubeConfig ensures the kube config is set to a valid file if provided.
+func IsValidKubeConfig(kubeconfig string) error {
+	if len(kubeconfig) > 0 {
+		if _, err := os.Stat(kubeconfig); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// IsValidPrometheusConfig ensures prometheus configuration is valid.
+func IsValidPrometheusConfig(cfg *Config) error {
+	errs := []error{}
+	if cfg.PrometheusConfig.CAFile != "" {
+		if _, err := os.Stat(cfg.PrometheusConfig.CAFile); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// PrometheusDataSourceMaxBackfillImportDuration overrides PrometheusDataSourceGlobalImportFromTime
+	// don't set both.
+	if cfg.PrometheusDataSourceGlobalImportFromTime != nil && cfg.PrometheusDataSourceMaxBackfillImportDuration > 0 {
+		errs = append(errs, fmt.Errorf("prometheusDataSourceGlobalImportFromTime and prometheusDataSourceMaxBackfillImportDuration cannot both be set"))
```

Comment:
previously we stated that one overrides the other.  Introducing this will cause explicit failure.  Acceptable?

Reply:
I'm not sure where I land on this validation check. I think it sounds reasonable looking at the codebase and making sure we reject configurations that won't be respected. @bentito any opinion on this validation check.

Reply:
@bentito - thoughts on this one?

Reply:
Reading the usage in `start.go`, it seems like this validates the stated usage. If it's Prom import question, they're both just ways of getting at how far back to try to grab Prom data when metering somehow gets out of sync. I think `PrometheusDataSourceGlobalImportFromTime` has more potential to be pretty awful for cluster workload, but I don't think that's being asked here.

Reply:
just validation of input. Sounds like I can leave it.  Thanks